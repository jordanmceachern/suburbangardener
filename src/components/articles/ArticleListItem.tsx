import React from "react";
import Link from "next/link";
import type { ArticleDisplayData } from "./ArticleDisplayCard";

interface ArticleListItemProps {
  article: ArticleDisplayData;
  showBorder?: boolean;
  className?: string;
}

export default function ArticleListItem({
  article,
  showBorder = true,
  className = "",
}: ArticleListItemProps) {
  return (
    <article
      className={`${
        showBorder ? "border-b border-slate-300 dark:border-slate-500 pb-4" : ""
      } ${className}`}
    >
      <h4 className="font-medium text-slate-800 dark:text-slate-200 mb-2">
        <Link
          href={`/articles/${article.slug}`}
          className="hover:text-primary-600 dark:hover:text-primary-300 transition-colors"
        >
          {article.title}
        </Link>
      </h4>
      <p className="text-sm text-slate-600 dark:text-slate-300 mb-2">
        {article.excerpt}
      </p>
      <span className="text-xs text-slate-500 dark:text-slate-300">
        {article.publishedDate}
      </span>
    </article>
  );
}
