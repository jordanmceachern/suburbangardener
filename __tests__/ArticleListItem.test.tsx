import { render } from "@testing-library/react";
import ArticleListItem from "@/components/articles/ArticleListItem";

describe("ArticleListItem Component", () => {
  const mockArticle = {
    id: "1",
    title: "Test Article",
    excerpt: "This is a test article excerpt",
    slug: "test-article",
    content: "Test content",
    imageUrl: "/test-image.jpg",
    author: "Test Author",
    publishedDate: "2024-01-01",
    category: "gardening",
  };

  it("should match DOM snapshot", () => {
    const { container } = render(<ArticleListItem article={mockArticle} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should maintain CSS classes for list item layout", () => {
    const { container } = render(<ArticleListItem article={mockArticle} />);

    // Check for list item classes
    const hasListClasses =
      container.innerHTML.includes("flex") ||
      container.innerHTML.includes("grid") ||
      container.innerHTML.includes("border") ||
      container.innerHTML.includes("hover:") ||
      container.innerHTML.includes("p-") ||
      container.innerHTML.includes("m-");

    expect(hasListClasses).toBe(true);
  });
});
