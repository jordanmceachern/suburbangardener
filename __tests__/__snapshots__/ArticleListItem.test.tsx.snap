// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ArticleListItem Component should match DOM snapshot 1`] = `
<article
  class="border-b border-slate-300 dark:border-slate-500 pb-4 "
>
  <h4
    class="font-medium text-slate-800 dark:text-slate-200 mb-2"
  >
    <a
      class="hover:text-primary-600 dark:hover:text-primary-300 transition-colors"
      href="/articles/test-article"
    >
      Test Article
    </a>
  </h4>
  <p
    class="text-sm text-slate-600 dark:text-slate-300 mb-2"
  >
    This is a test article excerpt
  </p>
  <span
    class="text-xs text-slate-500 dark:text-slate-300"
  >
    2024-01-01
  </span>
</article>
`;
